store the hltobjects for Mass50 triggers

pinchunchou opened the pull request:
#### PR description:
We've encountered an issue with the Mass50 HLT triggers for double electrons (Ele15Ele10GsfMass50, DoubleEle10GsfMass50, and DoubleEle15GsfMass50) in both ppref and HI HLT menus. These HLT triggers are not storing any information in the hltobject trees, even though the events are being triggered. After investigating, we have identified that the problem lies with HLTEgammaCombMassFilter.

I have made some modifications to HLTEgammaCombMassFilter, and now the information is being stored in hltobject.

See slides: https://twiki.cern.ch/twiki/pub/CMS/HLT_HIon_Run3/pinchun20230928.pdf (P.8)

Backport PR:
https://github.com/cms-sw/cmssw/pull/44746 (14_0_X)

pinchunchou committed:
store the hltobjects for Mass50 triggers

Add back a comment that I accidentally removed

Fix code-format

Fix per https://github.com/cms-sw/cmssw/pull/44746#discussion_r1571187288

Fix per https://github.com/cms-sw/cmssw/pull/44746#discussion_r1571187288

Only store the  trigger objects which pass the mass cut, fix per https://github.com/cms-sw/cmssw/pull/44746#discussion_r1571189315

code format

Change the default value of InputTag l1EGCand to hltEgammaCandidates, per https://github.com/cms-sw/cmssw/pull/44746#discussion_r1642253760

Fix l1EGCand customisation for HIon HLT menu using HLTEgammaCombMassFilter, per https://github.com/cms-sw/cmssw/pull/44746#discussion_r1645066715

fix

Fix per https://github.com/cms-sw/cmssw/pull/44746#discussion_r1647687130

cmsbuild commented:
cms-bot internal usage<!-- bot cache: {"commits":{"b4be6fa4d301d85b77cd07bdc9550b058492b908":{"files":["HLTrigger/Egamma/plugins/HLTEgammaCombMassFilter.cc","HLTrigger/Egamma/plugins/HLTEgammaCombMassFilter.h"],"squashed":false,"time":1713454652}},"emoji":{},"last_seen_sha":"b4be6fa4d301d85b77cd07bdc9550b058492b908","signatures":{}} -->

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-44778/40012

- This PR adds an extra 20KB to repository

Code check has found code style and quality issues which could be resolved by applying following patch(s)

- **code-format**:
https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-44778/40012/code-format.patch
e.g. `curl -k https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-44778/40012/code-format.patch | patch -p1`
You can also run `scram build code-format` to apply code format directly

mmusich commented:
@pinchunchou please apply the code formatting as per 
https://github.com/cms-sw/cmssw/pull/44778#issuecomment-2064779222

mmusich commented:
@pinchunchou this branch still fails the code checks. Please apply the code formatting suggested by the bot at https://github.com/cms-sw/cmssw/pull/44778#issuecomment-2064779222. Also there is a review in the 14.0.X version, can you please address it and port it also here?

Martin-Grunewald commented:
@pinchunchou this PR and its backport to 14_0 seem stuck, please address the comments and update the PRs.

Martin-Grunewald commented:
code-checks

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-44778/40147

- This PR adds an extra 12KB to repository

Code check has found code style and quality issues which could be resolved by applying following patch(s)

- **code-format**:
https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-44778/40147/code-format.patch
e.g. `curl -k https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-44778/40147/code-format.patch | patch -p1`
You can also run `scram build code-format` to apply code format directly

pinchunchou commented:
Hi @Martin-Grunewald and @mmusich , sorry for the late reply. I have applied the fix on code format in the newest commit. Thanks!

mmusich commented:
@pinchunchou, there is a review in the 14.0.X version https://github.com/cms-sw/cmssw/pull/44746, can you please address it and port it also here?

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-44778/40270

- This PR adds an extra 20KB to repository

cmsbuild commented:
A new Pull Request was created by  @pinchunchou for master.

It involves the following packages:

- HLTrigger/Egamma (**hlt**)


@mmusich, @Martin-Grunewald, @cmsbuild can you please review it and eventually sign? Thanks.
@silviodonato, @Martin-Grunewald, @missirol this is something you requested to watch as well.
@rappoccio, @sextonkennedy, @antoniovilela you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

mmusich commented:
@cmsbuild, please test (just to see the results)

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-2bc537/39560/summary.html
**COMMIT**: e3b170e55eaff11f2eb1bd064559f3e46d404a10
**CMSSW**: CMSSW_14_1_X_2024-05-27-2300/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/44778/39560/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - You potentially [removed 1 lines](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2024-05-27-2300+2bc537/62923/validateJR/logRootQA-events.log) from the logs
 - [Reco comparison results: 24 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2024-05-27-2300+2bc537/62923/validateJR.html)
 - DQMHistoTests: Total files compared: 48
 - DQMHistoTests: Total histograms compared: 3338862
 - DQMHistoTests: [Total failures: 886](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2024-05-27-2300+2bc537/62923/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3337956
 - DQMHistoTests: Total skipped: 20
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 47 files compared)
 - Checked 202 log files, 165 edm output root files, 48 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2024-05-27-2300+2bc537/62923/triggerResults): no differences found

Martin-Grunewald commented:
@pinchunchou, what is the status of this PR and its backport concerning the flagged updates?

mmusich commented:
-hlt
  * PR looks unattended
  * review at https://github.com/cms-sw/cmssw/pull/44746#pullrequestreview-2009586578 was never answered.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-44778/40597

- This PR adds an extra 20KB to repository

cmsbuild commented:
Pull request #44778 was updated. @mmusich, @cmsbuild, @Martin-Grunewald can you please check and sign again.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-44778/40618

- This PR adds an extra 24KB to repository

cmsbuild commented:
Pull request #44778 was updated. @mmusich, @cmsbuild, @Martin-Grunewald can you please check and sign again.

pinchunchou commented:
> @pinchunchou, what is the status of this PR and its backport concerning the flagged updates?

Hi Martin and all, I have just addressed the comments in its backport and modify both of them accordingly. Thanks for all your comments and suggestions.

mmusich commented:
@pinchunchou after you have fixed comment https://github.com/cms-sw/cmssw/pull/44746#discussion_r1571186803 please squash the commits to a minimum, only one if possible.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-44778/40643

- This PR adds an extra 20KB to repository

cmsbuild commented:
Pull request #44778 was updated. @cmsbuild, @mmusich, @Martin-Grunewald can you please check and sign again.

pinchunchou commented:
> @pinchunchou after you have fixed comment [#44746 (comment)](https://github.com/cms-sw/cmssw/pull/44746#discussion_r1571186803) please squash the commits to a minimum, only one if possible.

No problem! I have squashed them into a single commit.

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-44778/40646

ERROR: Unable to merge PR.

See log https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-44778/40646/cms-checkout-topic.log

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-44778/40647

ERROR: Unable to merge PR.

See log https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-44778/40647/cms-checkout-topic.log

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-44778/40648

- This PR adds an extra 28KB to repository

cmsbuild commented:
Pull request #44778 was updated. @cmsbuild, @mmusich, @Martin-Grunewald can you please check and sign again.

mmusich commented on HLTrigger/Configuration/python/customizeHLTforCMSSW.py:
```diff
@@ -243,7 +243,14 @@ def checkHLTfor43774(process):
     return process
-    
+
+def customizeHLTfor44746(process):
```
Technically this PR is 44778 and not 44746 (but I don't insist)

mmusich commented:
@cmsbuild, please test

mmusich commented:
@pinchunchou there are conflicts to solve.

cmsbuild commented:
-1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-2bc537/39979/summary.html
**COMMIT**: b3523174d07e55174af758a9c21203a1d715aa89
**CMSSW**: CMSSW_14_1_X_2024-06-19-1100/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/44778/39979/install.sh` to create a dev area with all the needed externals and cmssw changes.

This pull request cannot be automatically merged, could you please rebase it?
You can see the log for git cms-merge-topic here: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-2bc537/39979/git-merge-result

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-44778/40659

- This PR adds an extra 20KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  HLTrigger/Configuration/python/customizeHLTforCMSSW.py  modified in PR(s): #45206

cmsbuild commented:
Pull request #44778 was updated. @cmsbuild, @Martin-Grunewald, @mmusich can you please check and sign again.

mmusich commented:
@cmsbuild, please test

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-2bc537/39980/summary.html
**COMMIT**: e0fe125f7911b297bb83a580a87bf80a98517cb5
**CMSSW**: CMSSW_14_1_X_2024-06-19-2300/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/44778/39980/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - You potentially [added 6 lines](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2024-06-19-2300+2bc537/63283/validateJR/logRootQA-events.log) to the logs
 - [Reco comparison results: 4 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2024-06-19-2300+2bc537/63283/validateJR.html)
 - DQMHistoTests: Total files compared: 48
 - DQMHistoTests: Total histograms compared: 3345018
 - DQMHistoTests: [Total failures: 26](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2024-06-19-2300+2bc537/63283/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3344972
 - DQMHistoTests: Total skipped: 20
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 47 files compared)
 - Checked 202 log files, 165 edm output root files, 48 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2024-06-19-2300+2bc537/63283/triggerResults): no differences found

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-44778/40681

- This PR adds an extra 24KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  HLTrigger/Configuration/python/customizeHLTforCMSSW.py  modified in PR(s): #45206

cmsbuild commented:
Pull request #44778 was updated. @Martin-Grunewald, @cmsbuild, @mmusich can you please check and sign again.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-44778/40695

- This PR adds an extra 12KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  HLTrigger/Configuration/python/customizeHLTforCMSSW.py  modified in PR(s): #45206

cmsbuild commented:
Pull request #44778 was updated. @mmusich, @cmsbuild, @Martin-Grunewald can you please check and sign again.

Martin-Grunewald commented:
code-checks

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-44778/40747

ERROR: Unable to merge PR.

See log https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-44778/40747/cms-checkout-topic.log

Martin-Grunewald commented:
For this PR you'd need to fix the conflicts as shown above (the file with the customisation code has been changed by some other PR integrated in the meantime).

pinchunchou commented:
> For this PR you'd need to fix the conflicts as shown above (the file with the customisation code has been changed by some other PR integrated in the meantime).

Thanks for checking. Fixed

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-44778/40755

- This PR adds an extra 24KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  HLTrigger/Configuration/python/customizeHLTforCMSSW.py  modified in PR(s): #45326

cmsbuild commented:
Pull request #44778 was updated. @Martin-Grunewald, @cmsbuild, @mmusich can you please check and sign again.

Martin-Grunewald commented:
please test

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-2bc537/40149/summary.html
**COMMIT**: 4f67362574545c745d917c4b0bc944e62ed01445
**CMSSW**: CMSSW_14_1_X_2024-06-28-1100/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/44778/40149/install.sh` to create a dev area with all the needed externals and cmssw changes.


* **DAS Queries**: The DAS query tests failed, see the summary page for details.


## Comparison Summary



**Summary**:
 - You potentially [removed 3 lines](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2024-06-28-1100+2bc537/63417/validateJR/logRootQA-events.log) from the logs
 - [Reco comparison results: 2660 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2024-06-28-1100+2bc537/63417/validateJR.html)
 - DQMHistoTests: Total files compared: 48
 - DQMHistoTests: Total histograms compared: 3345088
 - DQMHistoTests: [Total failures: 1072](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2024-06-28-1100+2bc537/63417/validateJR.html)
 - DQMHistoTests: Total nulls: 4
 - DQMHistoTests: Total successes: 3343992
 - DQMHistoTests: Total skipped: 20
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 47 files compared)
 - Checked 202 log files, 165 edm output root files, 48 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2024-06-28-1100+2bc537/63417/triggerResults): found differences in 1 / 46 workflows

Martin-Grunewald commented:
please test

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-2bc537/40159/summary.html
**COMMIT**: 4f67362574545c745d917c4b0bc944e62ed01445
**CMSSW**: CMSSW_14_1_X_2024-06-30-2300/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/44778/40159/install.sh` to create a dev area with all the needed externals and cmssw changes.


* **DAS Queries**: The DAS query tests failed, see the summary page for details.


## Comparison Summary



**Summary**:
 - You potentially [removed 1 lines](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2024-06-30-2300+2bc537/63426/validateJR/logRootQA-events.log) from the logs
 - [Reco comparison results: 12 differences found in the comparisons](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2024-06-30-2300+2bc537/63426/validateJR.html)
 - DQMHistoTests: Total files compared: 48
 - DQMHistoTests: Total histograms compared: 3345088
 - DQMHistoTests: [Total failures: 35](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2024-06-30-2300+2bc537/63426/validateJR.html)
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3345033
 - DQMHistoTests: Total skipped: 20
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 47 files compared)
 - Checked 202 log files, 165 edm output root files, 48 DQM output files
 - [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_14_1_X_2024-06-30-2300+2bc537/63426/triggerResults): no differences found

Martin-Grunewald commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @rappoccio, @sextonkennedy, @antoniovilela (and backports should be raised in the release meeting by the corresponding L2)

antoniovilela commented:
+1
